os/arch/arm/src/imxrt/imxrt_qtmr.c

```diff
@@ -1,563 +1,999 @@
+		imxrt_qtmr_stoptimer(priv->qtmr->base, priv->channel);
+
+		if (active[priv->tmr] == 1) {
+			imxrt_qtmr_setisr(priv->qtmr, NULL, NULL);
+		}
+		priv->started = false;
+
+		busy_status[priv->tmr][priv->channel] = false;
+
+		active[priv->tmr] -= 1;
+		tmrvdbg("QTMR Timer %d channel %d is stopped\n", priv->tmr, priv->channel);
+		return OK;
+	}
+
+	/* Return ENODEV to indicate that the timer was not running */
+
+	tmrdbg("Error!! QTMR Timer %d channel %d is not running\n", priv->tmr, priv->channel);
+	return -ENODEV;
+}
+
+static int imxrt_qtmr_start(struct timer_lowerhalf_s *lower)
+{
+	struct imxrt_qtmr_lowerhalf_s *priv = (struct imxrt_qtmr_lowerhalf_s *)lower;
+	uint32_t enabled_interrupts;
+	uint16_t counter;
+	tmrvdbg("Start for timer[%d] channel [%d]\n", priv->tmr, priv->channel);
+
+	if (!priv->started) {
+		if (priv->callback != NULL) {
+			imxrt_qtmr_enableinterrupts(priv->qtmr->base, priv->channel, kQTMR_Compare1InterruptEnable);
+		}
+
+		enabled_interrupts = imxrt_qtmr_getenabledinterrupts(priv->qtmr->base, priv->channel);
+		enabled_interrupts &= kQTMR_Compare1InterruptEnable;
+		if (enabled_interrupts) {
+			tmrvdbg("enabled interrupt CMP1\n");
+		}
+
+		if (active[priv->tmr] == 0) {
+			imxrt_qtmr_setisr(priv->qtmr, imxrt_qtmr_handler, priv);
+		}
+
+		imxrt_qtmr_starttimer(priv->qtmr->base, priv->channel, kQTMR_PriSrcRiseEdge);
+
+		priv->started = true;
+		busy_status[priv->tmr][priv->channel] = true;
+
+		active[priv->tmr] += 1;
+
+		tmrvdbg("QTMR Timer %d Channel %d is started, callback %s\n", priv->tmr, priv->channel, priv->callback == NULL ? "none" : "set");
+
+		return OK;
+	}
+
+	/* Return EBUSY to indicate that the timer was already running */
+
+	tmrdbg("Error!! QTMR Timer %d Channel %d was already running\n", priv->tmr, priv->channel);
+	return -EBUSY;
+}
+
+int imxrt_qtmr_initialize(const char *devpath, int timer)
+{
+	qtmr_config_t config;
+	struct imxrt_qtmr_lowerhalf_s *lower = NULL;
+
+	/* At present, only one channel is being utilised */
+	int channel = 0;
+
+	switch (timer) {
+	case kQTMR_0:
+		lower = &g_qtmr0_0_lowerhalf;
+		break;
+
+	case kQTMR_1:
+		lower = &g_qtmr1_0_lowerhalf;
+		break;
+
+	case kQTMR_2:
+		lower = &g_qtmr2_0_lowerhalf;
+		break;
+
+	case kQTMR_3:
+		lower = &g_qtmr3_0_lowerhalf;
+		break;
+	}
+
+	if (!lower) {
+		return -ENODEV;
+	}
+
+	/* Initialize the elements of lower half state structure */
+	lower->started = false;
+	lower->callback = NULL;
+
+	imxrt_qtmr_getdefaultconfig(&config);
+	memcpy(&(lower->config), &config, sizeof(qtmr_config_t));
+	lower->qtmr = imxrt_qtmr_configure(timer, &config, channel);
+	lower->tmr = timer;
+	lower->channel = channel;
+
+	if (lower->qtmr == NULL) {
+		return -EINVAL;
+	}
+
+	/*
+	 * Register the timer driver as /dev/timerX.  The returned value from
+	 * timer_register is a handle that could be used with timer_unregister().
+	 * REVISIT: The returned handle is discard here.
+	 */
+	if (!timer_register(devpath, (struct timer_lowerhalf_s *)lower)) {
```

Comment:
This means some drivers will be created based on timer number.
So same number of iNode will be created. 
Each of iNode type is driver, and it will follow file operation.
But you try to change or refer to other iNode's priv values without open(path).
Please assume that each of different thread accessing all diffrent path at the same time. It's not stable.

Reply:
@Taejun-Kwon , I can't get your comment. I think one timer is not accessing any other timer's private data in the code.

Reply:
The channel configured by calling this function, based on number of channel, also each of driver registered in iNode.
It means that each iNode is kind of one file, also channel is configured when file is created.
So another file 'NEVER' knows another file, but you change other channle's config value.
In otherwords, you are doing like below
open file A -> modify data in the middle of file B without open(B) -> modify file A -> close A

Reply:
I mentioned it because your previous code changes each of channel's state(active...). 
Now you modified it then I think array active & busy_status is not necessary

Reply:
Yes...Thanks ...I have made the changes. I believe now active & busy_status are not required. Please check.